azurerm/config.go

```diff
@@ -189,7 +191,15 @@ func withRequestLogging() autorest.SendDecorator {
 
 func setUserAgent(client *autorest.Client) {
 	version := terraform.VersionString()
+	azureAgent := os.Getenv("AZURE_HTTP_USER_AGENT")
 	client.UserAgent = fmt.Sprintf("HashiCorp-Terraform-v%s", version)
+
+	if azureAgent != "" {
```

Comment:
_minor_ we'd tend to make this:

```
if azureAgent := os.Getenv("AZURE_HTTP_USER_AGENT"); azureAgent != "" {
 // set it
}
```

rather than separating it out :)